EIPS/eip-6672.md

```diff
@@ -0,0 +1,247 @@
+    /// @notice Redeem an NFT
+    /// @dev
+    /// @param _redemptionId The identifier created by the operator for a redemption.
+    /// @param _tokenId The NFT to redeem.
+    /// @param _memo
+    function redeem(bytes32 _redemptionId, uint256 _tokenId, string _memo) external;
+
+    /// @notice Cancel a redemption
+    /// @dev
+    /// @param _redemptionId The redemption to cancel.
+    /// @param _tokenId The NFT to cancel the redemption.
+    /// @param _memo
+    function cancel(bytes32 _redemptionId, uint256 _tokenId, string _memo) external;
+}
+```
+
+The **metadata extension** is OPTIONAL for [ERC-6672](./eip-6672.md) smart contracts (see "caveats", below). This allows your smart contract to be interrogated for its name and for details about the assets which your NFTs represent.
+
+```solidity
+/// @title ERC-6672 Multi-Redeemable Token Standard, optional metadata extension
+/// @dev See https://eips.ethereum.org/EIPS/eip-6672
+interface ERC6672Metadata /* is ERC721Metadata */ {
+    /// @notice A distinct Uniform Resource Identifier (URI) for a given asset.
+    /// @dev Throws if `_tokenId` is not a valid NFT. URIs are defined in RFC
+    ///  3986. The URI may point to a JSON file that conforms to the "ERC-6672
+    ///  Metadata JSON Schema".
+    function tokenURI(uint256 _tokenId) external view returns (string);
+}
+```
+
+This is the "[ERC-6672](./eip-6672.md) Metadata JSON Schema" referenced above.
+
+```json
+{
+    "title": "Asset Metadata",
+    "type": "object",
+    "properties": {
+        "name": {
+            "type": "string",
+            "description": "Identifies the asset to which this NFT represents"
+        },
+        "description": {
+            "type": "string",
+            "description": "Describes the asset to which this NFT represents"
+        },
+        "image": {
+            "type": "string",
+            "description": "A URI pointing to a resource with mime type image/* representing the asset to which this NFT represents. Consider making any images at a width between 320 and 1080 pixels and aspect ratio between 1.91:1 and 4:5 inclusive."
+        }
+    },
+    "redemptions": {
+        "operator-tokenId-redemptionId": {
+            "status": {
+                "type": "string",
+                "description": "The status of a redemption. Enum type can be used to represent the redemption status, such as redeemed, shipping, paid."
+            },
+            "description": {
+                "type": "string",
+                "description": "Describes the object that has been redeemed for an NFT, such as the name of an action figure series name or the color of the product."
+            }
+        }
+    }
+}
+```
+
+## Rationale
+
+### Redeem and Cancel Mechanism
+
+An operator SHALL only make an update to the redemption created by itself. Therefore, the `redeem()` and `cancel()` functions do not have an `_operator` parameter, and the `msg.sender` address MUST be used as the `_operator`.
```

Comment:
If "An operator SHALL only make an update to the redemption created by itself." is a requirement, it should go in the Specification section. The rest is good Rationale though.

Reply:
moved to Specification